src/vm/util.cpp

```diff
@@ -1645,36 +1645,21 @@ DWORD GetLogicalCpuCountFallback()
 
 #endif // _TARGET_X86_ || _TARGET_AMD64_
 
-// fix this if/when AMD does multicore or SMT
-size_t GetCacheSizePerLogicalCpu(BOOL bTrueSize)
+#if defined (_TARGET_X86_) || defined (_TARGET_AMD64_)
+static size_t GetCacheSizeFromCpuId()
 {
-    // No CONTRACT possible because GetCacheSizePerLogicalCpu uses SEH
-
     STATIC_CONTRACT_NOTHROW;
     STATIC_CONTRACT_GC_NOTRIGGER;
 
-    static size_t maxSize;
-    static size_t maxTrueSize;
+    // Can't return from a PAL_TRY. Instead, have it write to its parameter.
+    struct Param {
+        size_t maxSize;
+    } param = { 0 };
 
-    if (maxSize)
+    PAL_TRY(Param *, pParam, &param)
```

Comment:
Since the attached `PAL_EXCEPT_FILTER` is using `DefaultCatchFilter`, the parameter type has to be derived from `DefaultCatchFilterParam`. The `DefaultCatchFilter` casts the parameter * to `DefaultCatchFilterParam*` and reads the `pv` field out of it.
The `DefaultCatchFilter` is used to swallow hardware exceptions that can stem from the CPUID.